// @mindsorg/web3modal-angular/esm2015/lib/web3-modal/web3-modal.component.js
import { Component, Input, ViewEncapsulation, } from '@angular/core';
import { Web3ModalService } from './web3-modal.service';
export class Web3ModalComponent {
    constructor(service) {
        this.service = service;
        this.open = false;
        this.providers = [];
        this.metamaskShopURL = 'https://metamask.io/download.html';
        this.promptMetamaskIfNotInstalled = false;
    }
    ngOnInit() {
        this.openSubscription = this.service.shouldOpen.subscribe({
            next: (open) => {
                this.open = open;
            },
        });
        this.providersSubscription = this.service.providers.subscribe({
            next: (providers) => {
                this.showMetamaskDownload =
                    this.promptMetamaskIfNotInstalled &&
                        !this.isMetamaskInProviders(providers);
                this.providers = providers;
            },
        });
    }
    ngOnDestroy() {
        this.openSubscription.unsubscribe();
        this.providersSubscription.unsubscribe();
    }
    close() {
        this.service.close();
    }
    isMetamaskInProviders(providers) {
        return providers.some((p) => p.name.toLowerCase() === 'metamask');
    }
    openMetamaskDownloadPage() {
        window.open(this.metamaskShopURL, '_blank');
    }
}
Web3ModalComponent.decorators = [
    { type: Component, args: [{
                selector: 'm-web3-modal',
                template: "<m-modal [open]=\"open\" (closed)=\"close()\">\n  <div class=\"m-web3Modal\">\n    <div class=\"m-web3Modal__header\">\n      <h3 class=\"m-web3ModalHeader__title\">\n        {{ title }}\n      </h3>\n      <div class=\"m-web3ModalHeader__description\">\n        <p class=\"m-web3ModalHeader__text\">\n          {{ description }}\n        </p>\n        <p\n          class=\"m-web3ModalHeader__text--gray m-web3ModalHeader__text\"\n        >\n          {{ descriptionGray }}\n        </p>\n      </div>\n    </div>\n    <div class=\"m-web3Modal__body\">\n      <div class=\"m-web3ModalBody__provider\">\n        <div\n          class=\"m-web3ModalProvider__body\"\n          *ngIf=\"providers && showMetamaskDownload\"\n          (click)=\"openMetamaskDownloadPage()\"\n        >\n          <div class=\"m-web3ModalProviderBody__icon\">\n            <img src=\"metamask.svg\" alt=\"metamask.svg\" />\n          </div>\n          <div class=\"m-web3ModalProviderBody__name\">MetaMask</div>\n        </div>\n        <div\n          class=\"m-web3ModalProvider__body\"\n          *ngFor=\"let provider of providers\"\n          (click)=\"provider.onClick()\"\n        >\n          <div class=\"m-web3ModalProviderBody__icon\">\n            <img [src]=\"provider.logo\" [alt]=\"provider.logo\" />\n          </div>\n          <div class=\"m-web3ModalProviderBody__name\">\n            {{ provider.name }}\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"m-web3Modal__footer\">\n      <div class=\"m-web3ModalFooter__description\">\n        <p class=\"m-web3ModalFooter__text\" (click)=\"close()\">\n          {{ dismissText }}\n        </p>\n      </div>\n    </div>\n  </div>  \n</m-modal>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".m-web3Modal{position:relative;display:block;width:100%;margin:auto;box-sizing:border-box;background-color:#fff}@media screen and (max-width:540px){.m-web3Modal{flex:1;display:flex;flex-direction:column;justify-content:space-between;border-radius:0;margin:0}}.m-web3Modal__header{padding:40px 50px 0}.m-web3ModalHeader__title{font-size:30px;font-weight:800;margin:0;color:#4f4f50}.m-web3ModalHeader__text{margin:0;font-weight:600;line-height:22px;font-size:16px;color:#4f4f50}.m-web3ModalHeader__description{display:flex;flex-direction:column;justify-content:center;min-height:70px}.m-web3ModalHeader__text--gray{margin:0;color:#7d7d82}.m-web3Modal__footer{height:100px;margin-top:-40px;display:flex;padding:0 50px;flex-direction:column;justify-content:center}@media screen and (max-width:768px){.m-web3Modal__footer{margin-top:-20px}}.m-web3ModalFooter__text{text-align:center;font-size:17px;font-weight:600;margin:0;cursor:pointer;color:#7d7d82}.m-web3Modal__body{padding:20px 50px;max-height:420px;overflow-y:auto}.m-web3ModalBody__provider{display:flex;justify-content:space-between;flex-wrap:wrap}@media screen and (max-width:540px){.m-web3ModalBody__provider{justify-content:center}}.m-web3ModalProvider__body{display:flex;justify-content:center;align-items:center;flex-direction:column;cursor:pointer;width:194px;height:165px;margin-bottom:20px;background-color:#fff;border:1px solid #dce2e4}.m-web3ModalProviderBody__name{margin-top:12px;font-weight:600;font-size:15px;color:#4f4f50}.m-web3ModalProviderBody__icon{width:60px;height:60px;display:flex;border-radius:50%;overflow:visible;box-shadow:none;justify-content:center;align-items:center}.m-web3ModalProviderBody__icon img{width:100%;height:100%}"]
            },] }
];
Web3ModalComponent.ctorParameters = () => [
    { type: Web3ModalService }
];
Web3ModalComponent.propDecorators = {
    title: [{ type: Input }],
    description: [{ type: Input }],
    descriptionGray: [{ type: Input }],
    dismissText: [{ type: Input }],
    promptMetamaskIfNotInstalled: [{ type: Input }]
};
