// @mindsorg/web3modal-angular/__ivy_ngcc__/fesm2015/mindsorg-web3modal-angular.js
import { EventEmitter, Injectable, Optional, Component, ViewEncapsulation, Input, NgModule } from '@angular/core';
import { Web3WalletConnector, CONNECT_EVENT, ERROR_EVENT } from '@mindsorg/web3modal-ts';
import { take } from 'rxjs/operators';
import { CommonModule } from '@angular/common';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function Web3ModalComponent_div_12_Template(rf, ctx) { if (rf & 1) {
    const _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 14);
    ɵngcc0.ɵɵlistener("click", function Web3ModalComponent_div_12_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r3); const ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.openMetamaskDownloadPage(); });
    ɵngcc0.ɵɵelementStart(1, "div", 15);
    ɵngcc0.ɵɵelement(2, "img", 16);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div", 17);
    ɵngcc0.ɵɵtext(4, "MetaMask");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
function Web3ModalComponent_div_13_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 14);
    ɵngcc0.ɵɵlistener("click", function Web3ModalComponent_div_13_Template_div_click_0_listener() { const restoredCtx = ɵngcc0.ɵɵrestoreView(_r6); const provider_r4 = restoredCtx.$implicit; return provider_r4.onClick(); });
    ɵngcc0.ɵɵelementStart(1, "div", 15);
    ɵngcc0.ɵɵelement(2, "img", 18);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div", 17);
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const provider_r4 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("src", provider_r4.logo, ɵngcc0.ɵɵsanitizeUrl)("alt", provider_r4.logo);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", provider_r4.name, " ");
} }
const _c0 = ["*"];
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
class Web3ModalService {
    constructor(configOptions) {
        this.shouldOpen = new EventEmitter();
        this.providers = new EventEmitter();
        this.web3WalletConnector = new Web3WalletConnector(configOptions);
    }
    open() {
        return __awaiter(this, void 0, void 0, function* () {
            this.providers.next(this.web3WalletConnector.providers);
            return yield new Promise((resolve, reject) => {
                this.web3WalletConnector.providerController.on(CONNECT_EVENT, provider => {
                    resolve(provider);
                });
                this.web3WalletConnector.providerController.on(ERROR_EVENT, error => {
                    reject(error);
                });
                this.shouldOpen.next(true);
                this.shouldOpen.pipe(take(1)).subscribe({
                    next: (open) => {
                        if (!open) {
                            reject('Dismissed modal');
                        }
                    }
                });
            }).finally(() => {
                this.close();
            });
        });
    }
    setConfiguration(options) {
        this.web3WalletConnector.setConfiguration(options);
    }
    clearCachedProvider() {
        this.web3WalletConnector.providerController.clearCachedProvider();
    }
    setCachedProvider(id) {
        this.web3WalletConnector.providerController.setCachedProvider(id);
    }
    close() {
        this.shouldOpen.next(false);
    }
}
Web3ModalService.ɵfac = function Web3ModalService_Factory(t) { return new (t || Web3ModalService)(ɵngcc0.ɵɵinject(undefined, 8)); };
Web3ModalService.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: Web3ModalService, factory: Web3ModalService.ɵfac });
Web3ModalService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Optional }] }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Web3ModalService, [{
        type: Injectable
    }], function () { return [{ type: undefined, decorators: [{
                type: Optional
            }] }]; }, null); })();

class Web3ModalComponent {
    constructor(service) {
        this.service = service;
        this.open = false;
        this.providers = [];
        this.metamaskShopURL = 'https://metamask.io/download.html';
        this.promptMetamaskIfNotInstalled = false;
    }
    ngOnInit() {
        this.openSubscription = this.service.shouldOpen.subscribe({
            next: (open) => {
                this.open = open;
            },
        });
        this.providersSubscription = this.service.providers.subscribe({
            next: (providers) => {
                this.showMetamaskDownload =
                    this.promptMetamaskIfNotInstalled &&
                        !this.isMetamaskInProviders(providers);
                this.providers = providers;
            },
        });
    }
    ngOnDestroy() {
        this.openSubscription.unsubscribe();
        this.providersSubscription.unsubscribe();
    }
    close() {
        this.service.close();
    }
    isMetamaskInProviders(providers) {
        return providers.some((p) => p.name.toLowerCase() === 'metamask');
    }
    openMetamaskDownloadPage() {
        window.open(this.metamaskShopURL, '_blank');
    }
}
Web3ModalComponent.ɵfac = function Web3ModalComponent_Factory(t) { return new (t || Web3ModalComponent)(ɵngcc0.ɵɵdirectiveInject(Web3ModalService)); };
Web3ModalComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: Web3ModalComponent, selectors: [["m-web3-modal"]], inputs: { promptMetamaskIfNotInstalled: "promptMetamaskIfNotInstalled", title: "title", description: "description", descriptionGray: "descriptionGray", dismissText: "dismissText" }, decls: 18, vars: 7, consts: [[3, "open", "closed"], [1, "m-web3Modal"], [1, "m-web3Modal__header"], [1, "m-web3ModalHeader__title"], [1, "m-web3ModalHeader__description"], [1, "m-web3ModalHeader__text"], [1, "m-web3ModalHeader__text--gray", "m-web3ModalHeader__text"], [1, "m-web3Modal__body"], [1, "m-web3ModalBody__provider"], ["class", "m-web3ModalProvider__body", 3, "click", 4, "ngIf"], ["class", "m-web3ModalProvider__body", 3, "click", 4, "ngFor", "ngForOf"], [1, "m-web3Modal__footer"], [1, "m-web3ModalFooter__description"], [1, "m-web3ModalFooter__text", 3, "click"], [1, "m-web3ModalProvider__body", 3, "click"], [1, "m-web3ModalProviderBody__icon"], ["src", "metamask.svg", "alt", "metamask.svg"], [1, "m-web3ModalProviderBody__name"], [3, "src", "alt"]], template: function Web3ModalComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "m-modal", 0);
        ɵngcc0.ɵɵlistener("closed", function Web3ModalComponent_Template_m_modal_closed_0_listener() { return ctx.close(); });
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "h3", 3);
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 4);
        ɵngcc0.ɵɵelementStart(6, "p", 5);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "p", 6);
        ɵngcc0.ɵɵtext(9);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(10, "div", 7);
        ɵngcc0.ɵɵelementStart(11, "div", 8);
        ɵngcc0.ɵɵtemplate(12, Web3ModalComponent_div_12_Template, 5, 0, "div", 9);
        ɵngcc0.ɵɵtemplate(13, Web3ModalComponent_div_13_Template, 5, 3, "div", 10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "div", 11);
        ɵngcc0.ɵɵelementStart(15, "div", 12);
        ɵngcc0.ɵɵelementStart(16, "p", 13);
        ɵngcc0.ɵɵlistener("click", function Web3ModalComponent_Template_p_click_16_listener() { return ctx.close(); });
        ɵngcc0.ɵɵtext(17);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("open", ctx.open);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.title, " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.description, " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.descriptionGray, " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ctx.providers && ctx.showMetamaskDownload);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.providers);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.dismissText, " ");
    } }, directives: function () { return [Modal, ɵngcc1.NgIf, ɵngcc1.NgForOf]; }, styles: [".m-web3Modal{position:relative;display:block;width:100%;margin:auto;box-sizing:border-box;background-color:#fff}@media screen and (max-width:540px){.m-web3Modal{flex:1;display:flex;flex-direction:column;justify-content:space-between;border-radius:0;margin:0}}.m-web3Modal__header{padding:40px 50px 0}.m-web3ModalHeader__title{font-size:30px;font-weight:800;margin:0;color:#4f4f50}.m-web3ModalHeader__text{margin:0;font-weight:600;line-height:22px;font-size:16px;color:#4f4f50}.m-web3ModalHeader__description{display:flex;flex-direction:column;justify-content:center;min-height:70px}.m-web3ModalHeader__text--gray{margin:0;color:#7d7d82}.m-web3Modal__footer{height:100px;margin-top:-40px;display:flex;padding:0 50px;flex-direction:column;justify-content:center}@media screen and (max-width:768px){.m-web3Modal__footer{margin-top:-20px}}.m-web3ModalFooter__text{text-align:center;font-size:17px;font-weight:600;margin:0;cursor:pointer;color:#7d7d82}.m-web3Modal__body{padding:20px 50px;max-height:420px;overflow-y:auto}.m-web3ModalBody__provider{display:flex;justify-content:space-between;flex-wrap:wrap}@media screen and (max-width:540px){.m-web3ModalBody__provider{justify-content:center}}.m-web3ModalProvider__body{display:flex;justify-content:center;align-items:center;flex-direction:column;cursor:pointer;width:194px;height:165px;margin-bottom:20px;background-color:#fff;border:1px solid #dce2e4}.m-web3ModalProviderBody__name{margin-top:12px;font-weight:600;font-size:15px;color:#4f4f50}.m-web3ModalProviderBody__icon{width:60px;height:60px;display:flex;border-radius:50%;overflow:visible;box-shadow:none;justify-content:center;align-items:center}.m-web3ModalProviderBody__icon img{width:100%;height:100%}"], encapsulation: 2 });
Web3ModalComponent.ctorParameters = () => [
    { type: Web3ModalService }
];
Web3ModalComponent.propDecorators = {
    title: [{ type: Input }],
    description: [{ type: Input }],
    descriptionGray: [{ type: Input }],
    dismissText: [{ type: Input }],
    promptMetamaskIfNotInstalled: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Web3ModalComponent, [{
        type: Component,
        args: [{
                selector: 'm-web3-modal',
                template: "<m-modal [open]=\"open\" (closed)=\"close()\">\n  <div class=\"m-web3Modal\">\n    <div class=\"m-web3Modal__header\">\n      <h3 class=\"m-web3ModalHeader__title\">\n        {{ title }}\n      </h3>\n      <div class=\"m-web3ModalHeader__description\">\n        <p class=\"m-web3ModalHeader__text\">\n          {{ description }}\n        </p>\n        <p\n          class=\"m-web3ModalHeader__text--gray m-web3ModalHeader__text\"\n        >\n          {{ descriptionGray }}\n        </p>\n      </div>\n    </div>\n    <div class=\"m-web3Modal__body\">\n      <div class=\"m-web3ModalBody__provider\">\n        <div\n          class=\"m-web3ModalProvider__body\"\n          *ngIf=\"providers && showMetamaskDownload\"\n          (click)=\"openMetamaskDownloadPage()\"\n        >\n          <div class=\"m-web3ModalProviderBody__icon\">\n            <img src=\"metamask.svg\" alt=\"metamask.svg\" />\n          </div>\n          <div class=\"m-web3ModalProviderBody__name\">MetaMask</div>\n        </div>\n        <div\n          class=\"m-web3ModalProvider__body\"\n          *ngFor=\"let provider of providers\"\n          (click)=\"provider.onClick()\"\n        >\n          <div class=\"m-web3ModalProviderBody__icon\">\n            <img [src]=\"provider.logo\" [alt]=\"provider.logo\" />\n          </div>\n          <div class=\"m-web3ModalProviderBody__name\">\n            {{ provider.name }}\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"m-web3Modal__footer\">\n      <div class=\"m-web3ModalFooter__description\">\n        <p class=\"m-web3ModalFooter__text\" (click)=\"close()\">\n          {{ dismissText }}\n        </p>\n      </div>\n    </div>\n  </div>  \n</m-modal>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".m-web3Modal{position:relative;display:block;width:100%;margin:auto;box-sizing:border-box;background-color:#fff}@media screen and (max-width:540px){.m-web3Modal{flex:1;display:flex;flex-direction:column;justify-content:space-between;border-radius:0;margin:0}}.m-web3Modal__header{padding:40px 50px 0}.m-web3ModalHeader__title{font-size:30px;font-weight:800;margin:0;color:#4f4f50}.m-web3ModalHeader__text{margin:0;font-weight:600;line-height:22px;font-size:16px;color:#4f4f50}.m-web3ModalHeader__description{display:flex;flex-direction:column;justify-content:center;min-height:70px}.m-web3ModalHeader__text--gray{margin:0;color:#7d7d82}.m-web3Modal__footer{height:100px;margin-top:-40px;display:flex;padding:0 50px;flex-direction:column;justify-content:center}@media screen and (max-width:768px){.m-web3Modal__footer{margin-top:-20px}}.m-web3ModalFooter__text{text-align:center;font-size:17px;font-weight:600;margin:0;cursor:pointer;color:#7d7d82}.m-web3Modal__body{padding:20px 50px;max-height:420px;overflow-y:auto}.m-web3ModalBody__provider{display:flex;justify-content:space-between;flex-wrap:wrap}@media screen and (max-width:540px){.m-web3ModalBody__provider{justify-content:center}}.m-web3ModalProvider__body{display:flex;justify-content:center;align-items:center;flex-direction:column;cursor:pointer;width:194px;height:165px;margin-bottom:20px;background-color:#fff;border:1px solid #dce2e4}.m-web3ModalProviderBody__name{margin-top:12px;font-weight:600;font-size:15px;color:#4f4f50}.m-web3ModalProviderBody__icon{width:60px;height:60px;display:flex;border-radius:50%;overflow:visible;box-shadow:none;justify-content:center;align-items:center}.m-web3ModalProviderBody__icon img{width:100%;height:100%}"]
            }]
    }], function () { return [{ type: Web3ModalService }]; }, { promptMetamaskIfNotInstalled: [{
            type: Input
        }], title: [{
            type: Input
        }], description: [{
            type: Input
        }], descriptionGray: [{
            type: Input
        }], dismissText: [{
            type: Input
        }] }); })();

class Modal {
    constructor() {
        this.allowClose = true;
        this.hidden = true;
        this.closed = new EventEmitter();
    }
    set _hidden(value) {
        this.hidden = value;
    }
    set open(value) {
        this.hidden = !value;
    }
    close(event) {
        if (!this.allowClose)
            return;
        this.hidden = !this.hidden;
        this.closed.next(true);
        event.stopPropagation();
    }
}
Modal.ɵfac = function Modal_Factory(t) { return new (t || Modal)(); };
Modal.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: Modal, selectors: [["m-modal"]], hostVars: 1, hostBindings: function Modal_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵhostProperty("hidden", ctx.hidden);
    } }, inputs: { open: "open", allowClose: "allowClose" }, outputs: { closed: "closed" }, ngContentSelectors: _c0, decls: 3, vars: 0, consts: [[1, "m-modal-bg", 3, "click"], [1, "m-modal-container"]], template: function Modal_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("click", function Modal_Template_div_click_0_listener($event) { return ctx.close($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementEnd();
    } }, styles: ["m-modal .m-modal-bg{width:100%;height:100%;top:0;left:0;position:fixed;cursor:pointer;z-index:99999991;background-color:rgba(0,0,0,.8)}m-modal .m-modal-container{position:fixed;margin:auto;z-index:99999995;width:800px;max-width:100%;left:50%;top:50%;max-height:98vh;display:block;box-sizing:border-box;transform:translate(-50%,-50%);padding:32px;outline:0;border-radius:6px;overflow-y:auto;-webkit-overflow-scrolling:touch;font-family:Roboto,sans-serif}m-modal{position:relative;top:0;bottom:0;left:0;right:0;height:100%;width:100%;overflow:overlay;overflow:scroll}"], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Modal, [{
        type: Component,
        args: [{
                selector: 'm-modal',
                host: {
                    '[hidden]': 'hidden'
                },
                inputs: ['open', 'allowClose'],
                outputs: ['closed'],
                encapsulation: ViewEncapsulation.None,
                template: "<div class=\"m-modal-bg\" (click)=\"close($event)\"></div>\n<div class=\"m-modal-container\">\n  <ng-content></ng-content>\n</div>",
                styles: ["m-modal .m-modal-bg{width:100%;height:100%;top:0;left:0;position:fixed;cursor:pointer;z-index:99999991;background-color:rgba(0,0,0,.8)}m-modal .m-modal-container{position:fixed;margin:auto;z-index:99999995;width:800px;max-width:100%;left:50%;top:50%;max-height:98vh;display:block;box-sizing:border-box;transform:translate(-50%,-50%);padding:32px;outline:0;border-radius:6px;overflow-y:auto;-webkit-overflow-scrolling:touch;font-family:Roboto,sans-serif}m-modal{position:relative;top:0;bottom:0;left:0;right:0;height:100%;width:100%;overflow:overlay;overflow:scroll}"]
            }]
    }], function () { return []; }, null); })();

class Web3ModalModule {
}
Web3ModalModule.ɵfac = function Web3ModalModule_Factory(t) { return new (t || Web3ModalModule)(); };
Web3ModalModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: Web3ModalModule });
Web3ModalModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Web3ModalModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [Modal, Web3ModalComponent],
                exports: [Web3ModalComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(Web3ModalModule, { declarations: function () { return [Modal, Web3ModalComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [Web3ModalComponent]; } }); })();

/*
 * Public API Surface of web3modal
 */

/**
 * Generated bundle index. Do not edit.
 */

export { Web3ModalComponent, Web3ModalModule, Web3ModalService, Modal as ɵa };
