// @mindsorg/web3modal-angular/bundles/mindsorg-web3modal-angular.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@mindsorg/web3modal-ts'), require('rxjs/operators'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@mindsorg/web3modal-angular', ['exports', '@angular/core', '@mindsorg/web3modal-ts', 'rxjs/operators', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.mindsorg = global.mindsorg || {}, global.mindsorg['web3modal-angular'] = {}), global.ng.core, global.web3modalTs, global.rxjs.operators, global.ng.common));
}(this, (function (exports, core, web3modalTs, operators, common) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter$1(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || from);
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    };
    var Web3ModalService = /** @class */ (function () {
        function Web3ModalService(configOptions) {
            this.shouldOpen = new core.EventEmitter();
            this.providers = new core.EventEmitter();
            this.web3WalletConnector = new web3modalTs.Web3WalletConnector(configOptions);
        }
        Web3ModalService.prototype.open = function () {
            return __awaiter(this, void 0, void 0, function () {
                var _this = this;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            this.providers.next(this.web3WalletConnector.providers);
                            return [4 /*yield*/, new Promise(function (resolve, reject) {
                                    _this.web3WalletConnector.providerController.on(web3modalTs.CONNECT_EVENT, function (provider) {
                                        resolve(provider);
                                    });
                                    _this.web3WalletConnector.providerController.on(web3modalTs.ERROR_EVENT, function (error) {
                                        reject(error);
                                    });
                                    _this.shouldOpen.next(true);
                                    _this.shouldOpen.pipe(operators.take(1)).subscribe({
                                        next: function (open) {
                                            if (!open) {
                                                reject('Dismissed modal');
                                            }
                                        }
                                    });
                                }).finally(function () {
                                    _this.close();
                                })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        Web3ModalService.prototype.setConfiguration = function (options) {
            this.web3WalletConnector.setConfiguration(options);
        };
        Web3ModalService.prototype.clearCachedProvider = function () {
            this.web3WalletConnector.providerController.clearCachedProvider();
        };
        Web3ModalService.prototype.setCachedProvider = function (id) {
            this.web3WalletConnector.providerController.setCachedProvider(id);
        };
        Web3ModalService.prototype.close = function () {
            this.shouldOpen.next(false);
        };
        return Web3ModalService;
    }());
    Web3ModalService.decorators = [
        { type: core.Injectable }
    ];
    Web3ModalService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: core.Optional }] }
    ]; };

    var Web3ModalComponent = /** @class */ (function () {
        function Web3ModalComponent(service) {
            this.service = service;
            this.open = false;
            this.providers = [];
            this.metamaskShopURL = 'https://metamask.io/download.html';
            this.promptMetamaskIfNotInstalled = false;
        }
        Web3ModalComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.openSubscription = this.service.shouldOpen.subscribe({
                next: function (open) {
                    _this.open = open;
                },
            });
            this.providersSubscription = this.service.providers.subscribe({
                next: function (providers) {
                    _this.showMetamaskDownload =
                        _this.promptMetamaskIfNotInstalled &&
                            !_this.isMetamaskInProviders(providers);
                    _this.providers = providers;
                },
            });
        };
        Web3ModalComponent.prototype.ngOnDestroy = function () {
            this.openSubscription.unsubscribe();
            this.providersSubscription.unsubscribe();
        };
        Web3ModalComponent.prototype.close = function () {
            this.service.close();
        };
        Web3ModalComponent.prototype.isMetamaskInProviders = function (providers) {
            return providers.some(function (p) { return p.name.toLowerCase() === 'metamask'; });
        };
        Web3ModalComponent.prototype.openMetamaskDownloadPage = function () {
            window.open(this.metamaskShopURL, '_blank');
        };
        return Web3ModalComponent;
    }());
    Web3ModalComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'm-web3-modal',
                    template: "<m-modal [open]=\"open\" (closed)=\"close()\">\n  <div class=\"m-web3Modal\">\n    <div class=\"m-web3Modal__header\">\n      <h3 class=\"m-web3ModalHeader__title\">\n        {{ title }}\n      </h3>\n      <div class=\"m-web3ModalHeader__description\">\n        <p class=\"m-web3ModalHeader__text\">\n          {{ description }}\n        </p>\n        <p\n          class=\"m-web3ModalHeader__text--gray m-web3ModalHeader__text\"\n        >\n          {{ descriptionGray }}\n        </p>\n      </div>\n    </div>\n    <div class=\"m-web3Modal__body\">\n      <div class=\"m-web3ModalBody__provider\">\n        <div\n          class=\"m-web3ModalProvider__body\"\n          *ngIf=\"providers && showMetamaskDownload\"\n          (click)=\"openMetamaskDownloadPage()\"\n        >\n          <div class=\"m-web3ModalProviderBody__icon\">\n            <img src=\"metamask.svg\" alt=\"metamask.svg\" />\n          </div>\n          <div class=\"m-web3ModalProviderBody__name\">MetaMask</div>\n        </div>\n        <div\n          class=\"m-web3ModalProvider__body\"\n          *ngFor=\"let provider of providers\"\n          (click)=\"provider.onClick()\"\n        >\n          <div class=\"m-web3ModalProviderBody__icon\">\n            <img [src]=\"provider.logo\" [alt]=\"provider.logo\" />\n          </div>\n          <div class=\"m-web3ModalProviderBody__name\">\n            {{ provider.name }}\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"m-web3Modal__footer\">\n      <div class=\"m-web3ModalFooter__description\">\n        <p class=\"m-web3ModalFooter__text\" (click)=\"close()\">\n          {{ dismissText }}\n        </p>\n      </div>\n    </div>\n  </div>  \n</m-modal>\n",
                    encapsulation: core.ViewEncapsulation.None,
                    styles: [".m-web3Modal{position:relative;display:block;width:100%;margin:auto;box-sizing:border-box;background-color:#fff}@media screen and (max-width:540px){.m-web3Modal{flex:1;display:flex;flex-direction:column;justify-content:space-between;border-radius:0;margin:0}}.m-web3Modal__header{padding:40px 50px 0}.m-web3ModalHeader__title{font-size:30px;font-weight:800;margin:0;color:#4f4f50}.m-web3ModalHeader__text{margin:0;font-weight:600;line-height:22px;font-size:16px;color:#4f4f50}.m-web3ModalHeader__description{display:flex;flex-direction:column;justify-content:center;min-height:70px}.m-web3ModalHeader__text--gray{margin:0;color:#7d7d82}.m-web3Modal__footer{height:100px;margin-top:-40px;display:flex;padding:0 50px;flex-direction:column;justify-content:center}@media screen and (max-width:768px){.m-web3Modal__footer{margin-top:-20px}}.m-web3ModalFooter__text{text-align:center;font-size:17px;font-weight:600;margin:0;cursor:pointer;color:#7d7d82}.m-web3Modal__body{padding:20px 50px;max-height:420px;overflow-y:auto}.m-web3ModalBody__provider{display:flex;justify-content:space-between;flex-wrap:wrap}@media screen and (max-width:540px){.m-web3ModalBody__provider{justify-content:center}}.m-web3ModalProvider__body{display:flex;justify-content:center;align-items:center;flex-direction:column;cursor:pointer;width:194px;height:165px;margin-bottom:20px;background-color:#fff;border:1px solid #dce2e4}.m-web3ModalProviderBody__name{margin-top:12px;font-weight:600;font-size:15px;color:#4f4f50}.m-web3ModalProviderBody__icon{width:60px;height:60px;display:flex;border-radius:50%;overflow:visible;box-shadow:none;justify-content:center;align-items:center}.m-web3ModalProviderBody__icon img{width:100%;height:100%}"]
                },] }
    ];
    Web3ModalComponent.ctorParameters = function () { return [
        { type: Web3ModalService }
    ]; };
    Web3ModalComponent.propDecorators = {
        title: [{ type: core.Input }],
        description: [{ type: core.Input }],
        descriptionGray: [{ type: core.Input }],
        dismissText: [{ type: core.Input }],
        promptMetamaskIfNotInstalled: [{ type: core.Input }]
    };

    var Modal = /** @class */ (function () {
        function Modal() {
            this.allowClose = true;
            this.hidden = true;
            this.closed = new core.EventEmitter();
        }
        Object.defineProperty(Modal.prototype, "_hidden", {
            set: function (value) {
                this.hidden = value;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(Modal.prototype, "open", {
            set: function (value) {
                this.hidden = !value;
            },
            enumerable: false,
            configurable: true
        });
        Modal.prototype.close = function (event) {
            if (!this.allowClose)
                return;
            this.hidden = !this.hidden;
            this.closed.next(true);
            event.stopPropagation();
        };
        return Modal;
    }());
    Modal.decorators = [
        { type: core.Component, args: [{
                    selector: 'm-modal',
                    host: {
                        '[hidden]': 'hidden',
                    },
                    inputs: ['open', 'allowClose'],
                    outputs: ['closed'],
                    encapsulation: core.ViewEncapsulation.None,
                    template: "<div class=\"m-modal-bg\" (click)=\"close($event)\"></div>\n<div class=\"m-modal-container\">\n  <ng-content></ng-content>\n</div>",
                    styles: ["m-modal .m-modal-bg{width:100%;height:100%;top:0;left:0;position:fixed;cursor:pointer;z-index:99999991;background-color:rgba(0,0,0,.8)}m-modal .m-modal-container{position:fixed;margin:auto;z-index:99999995;width:800px;max-width:100%;left:50%;top:50%;max-height:98vh;display:block;box-sizing:border-box;transform:translate(-50%,-50%);padding:32px;outline:0;border-radius:6px;overflow-y:auto;-webkit-overflow-scrolling:touch;font-family:Roboto,sans-serif}m-modal{position:relative;top:0;bottom:0;left:0;right:0;height:100%;width:100%;overflow:overlay;overflow:scroll}"]
                },] }
    ];

    var Web3ModalModule = /** @class */ (function () {
        function Web3ModalModule() {
        }
        return Web3ModalModule;
    }());
    Web3ModalModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [common.CommonModule],
                    declarations: [Modal, Web3ModalComponent],
                    exports: [Web3ModalComponent],
                },] }
    ];

    /*
     * Public API Surface of web3modal
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.Web3ModalComponent = Web3ModalComponent;
    exports.Web3ModalModule = Web3ModalModule;
    exports.Web3ModalService = Web3ModalService;
    exports.ɵa = Modal;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
