// @mindsorg/web3modal-angular/esm2015/lib/web3-modal/web3-modal.service.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { EventEmitter, Injectable, Optional } from '@angular/core';
import { CONNECT_EVENT, ERROR_EVENT, Web3WalletConnector } from '@mindsorg/web3modal-ts';
import { take } from 'rxjs/operators';
export class Web3ModalService {
    constructor(configOptions) {
        this.shouldOpen = new EventEmitter();
        this.providers = new EventEmitter();
        this.web3WalletConnector = new Web3WalletConnector(configOptions);
    }
    open() {
        return __awaiter(this, void 0, void 0, function* () {
            this.providers.next(this.web3WalletConnector.providers);
            return yield new Promise((resolve, reject) => {
                this.web3WalletConnector.providerController.on(CONNECT_EVENT, provider => {
                    resolve(provider);
                });
                this.web3WalletConnector.providerController.on(ERROR_EVENT, error => {
                    reject(error);
                });
                this.shouldOpen.next(true);
                this.shouldOpen.pipe(take(1)).subscribe({
                    next: (open) => {
                        if (!open) {
                            reject('Dismissed modal');
                        }
                    }
                });
            }).finally(() => {
                this.close();
            });
        });
    }
    setConfiguration(options) {
        this.web3WalletConnector.setConfiguration(options);
    }
    clearCachedProvider() {
        this.web3WalletConnector.providerController.clearCachedProvider();
    }
    setCachedProvider(id) {
        this.web3WalletConnector.providerController.setCachedProvider(id);
    }
    close() {
        this.shouldOpen.next(false);
    }
}
Web3ModalService.decorators = [
    { type: Injectable }
];
Web3ModalService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Optional }] }
];
