// @mindsorg/web3modal-ts/dist/providers/providers/index.js
export * from '../injected';
export const WALLETCONNECT = {
    id: 'walletconnect',
    name: 'WalletConnect',
    logo: new URL('../logos/walletconnect-circle.svg', import.meta.url),
    type: 'qrcode',
    check: 'isWalletConnect',
    package: {
        required: [['infuraId', 'rpc']],
    },
};
export const WALLETLINK = {
    id: 'walletlink',
    name: 'Coinbase',
    logo: new URL('../logos/coinbase.svg', import.meta.url),
    type: 'qrcode',
    check: 'isWalletLink',
    package: {
        required: ['infuraUrl'],
    },
};
export const PORTIS = {
    id: 'portis',
    name: 'Portis',
    logo: new URL('../logos/portis.svg', import.meta.url),
    type: 'web',
    check: 'isPortis',
    package: {
        required: ['id'],
    },
};
export const FORTMATIC = {
    id: 'fortmatic',
    name: 'Fortmatic',
    logo: new URL('../logos/fortmatic.svg', import.meta.url),
    type: 'web',
    check: 'isFortmatic',
    package: {
        required: ['key'],
    },
};
export const SQUARELINK = {
    id: 'squarelink',
    name: 'Squarelink',
    logo: new URL('../logos/squarelink.svg', import.meta.url),
    type: 'web',
    check: 'isSquarelink',
    package: {
        required: ['id'],
    },
};
export const TORUS = {
    id: 'torus',
    name: 'Torus',
    logo: new URL('../logos/torus.svg', import.meta.url),
    type: 'web',
    check: 'isTorus',
};
export const ARKANE = {
    id: 'arkane',
    name: 'Arkane',
    logo: new URL('../logos/arkane.svg', import.meta.url),
    type: 'web',
    check: 'isArkane',
    package: {
        required: ['clientId'],
    },
};
export const AUTHEREUM = {
    id: 'authereum',
    name: 'Authereum',
    logo: new URL('../logos/authereum.svg', import.meta.url),
    type: 'web',
    check: 'isAuthereum',
};
export const BURNERCONNECT = {
    id: 'burnerconnect',
    name: 'Burner Connect',
    logo: new URL('../logos/burnerwallet.png', import.meta.url),
    type: 'web',
    check: 'isBurnerProvider',
};
export const UNILOGIN = {
    id: 'unilogin',
    name: 'UniLogin',
    logo: new URL('../logos/unilogin.svg', import.meta.url),
    check: 'isUniLogin',
    type: 'web',
};
export const MEWCONNECT = {
    id: 'mewconnect',
    name: 'MEW wallet',
    logo: new URL('../logos/mewwallet.png', import.meta.url),
    type: 'qrcode',
    check: 'isMEWconnect',
    package: {
        required: [['infuraId', 'rpc']],
    },
};
export const DCENT = {
    id: 'dcentwallet',
    name: "D'CENT",
    logo: new URL('../logos/dcentwallet.png', import.meta.url),
    type: 'hardware',
    check: 'isDcentWallet',
    package: {
        required: ['rpcUrl'],
    },
};
export const BITSKI = {
    id: 'bitski',
    name: 'Bitski',
    logo: new URL('../logos/bitski.svg', import.meta.url),
    type: 'web',
    check: 'isBitski',
    package: {
        required: ['clientId', 'callbackUrl'],
    },
};
